From 0dcb41bd49b7d1d15e7d25b659a0d69ee550bae3 Mon Sep 17 00:00:00 2001
From: Michael Tautschnig <mt@debian.org>
Date: Wed, 31 May 2017 09:29:03 +0000
Subject: [PATCH] Ensure consistent typing of aceInit

gcc -g -Wall  -DACEDB4 `../w3rdparty/include-config glib-2.0 gtk+-2.0` -g -O0 -fdebug-prefix-map=/home/admin/workspace/sid-goto-cc-acedb/acedb-4.9.39+dfsg.02=. -fstack-protector-strong -Wformat -Werror=format-security  -I.. -I../wh -I../wstaden -DACEDB_GTK -DLINUX -c -o metadata.o metadata.c
metadata.c: In function 'main':
metadata.c:190:5: warning: implicit declaration of function 'aceInit' [-Wimplicit-function-declaration]
     aceInit(argv[1]);
     ^~~~~~~
metadata.c: In function 'main':
metadata.c:190:1: warning: function `aceInit' is not declared
gcc -g -Wl,-z,relro -Wl,-z,now -o metacheck metadata.o -L. -lace -lfree -L/usr/X11R6/lib `../w3rdparty/libs-config glib-2.0 gtk+-2.0` -lm -lreadline -lX11 -lfl
./libfree.a(filsubs.o): In function `filTmpOpenWithSuffix':
./bin.LINUX_4/filsubs.c:588: warning: the use of `tempnam' is dangerous, better use `mkstemp'
../wh/acelib.h:168:1: error:
reason for conflict at #this: type classes differ

enum  { FALSE=0, TRUE=1 }
Bool
error: conflicting function declarations `aceInit'
old definition in module `metadata' file acelib.c line 149
BOOL (char *ace_path)
new definition in module `acinside' file ../wh/acelib.h line 168
Bool (char *)
makefile:1279: recipe for target 'metacheck' failed
make[3]: *** [metacheck] Error 1
---
 w4/metadata.c | 1 +
 wh/acetypes.h | 9 +--------
 2 files changed, 2 insertions(+), 8 deletions(-)

diff --git a/w4/metadata.c b/w4/metadata.c
index d817028..80878d1 100644
--- a/w4/metadata.c
+++ b/w4/metadata.c
@@ -37,6 +37,7 @@
 
 /* $Id: metadata.c,v 1.14 2003/07/09 13:20:25 rnc Exp $ */
 
+#include "acelib.h"
 #include "acedb.h"
 
 #include "lex.h"
diff --git a/wh/acetypes.h b/wh/acetypes.h
index 78a77e9..64f0459 100644
--- a/wh/acetypes.h
+++ b/wh/acetypes.h
@@ -43,16 +43,9 @@ typedef enum
 #ifndef  ACELIB__H   /* private kernel definitions */
 
 typedef void  AceContext ;
-typedef int   Bool;
-
-#ifndef TRUE
-#define TRUE 1
-#endif
-#ifndef FALSE
-#define FALSE 0
-#endif
 
 #include "regular.h"
+typedef BOOL Bool;
 #include "mytime.h"		/* not included in standard libutil header */
 
 #include "keyset.h"
-- 
2.1.4

